File: NearbyPlaces/UIImage+scaleImage.m
//
//  UIImage+scaleImage.m
//  NearbyPlaces
//
//  Created by Adrian Kubała on 14.02.2017.
//  Copyright © 2017 Adrian Kubała. All rights reserved.
//

#import "UIImage+scaleImage.h"

@implementation UIImage (scaleImage)

- (instancetype)scaleImage:(double)width {
  CGFloat scale = width / self.size.width;
  CGFloat newHeight = self.size.height * scale;
  
  UIGraphicsBeginImageContext(CGSizeMake(width, newHeight));
  [self drawInRect:CGRectMake(0, 0, width, newHeight)];
  UIImage *newImage = UIGraphicsGetImageFromCurrentImageContext();
  UIGraphicsEndImageContext();
  
  return newImage;
}

@end

File: Places/UIImage+scaleImage.swift
//
//  UIImage+scaleImage.swift
//  Places
//
//  Created by Adrian on 29.09.2016.
//  Copyright © 2016 Adrian Kubała. All rights reserved.
//

import UIKit

extension UIImage {
  func scaleImage(width: Double) -> UIImage {
    let newWidth = CGFloat(width)
    let scale = newWidth / size.width
    let newHeight = size.height * scale
    UIGraphicsBeginImageContext(CGSize(width: newWidth, height: newHeight))
    draw(in: CGRect(x: 0, y: 0, width: newWidth, height: newHeight))
    let newImage = UIGraphicsGetImageFromCurrentImageContext()
    UIGraphicsEndImageContext()
    
    return newImage!
  }
}
